### PR TITLE
Bugfix for LOG4J2-3107: SmtpManager.createManagerName ignores port

### PR DESCRIPTION
**Pull request by sman-81**
My fix for LOG4J2-3107 changes two files and adds one. The change is now targeted at the release-2.x branch not master.
Hope you find it useful.

### CODE REVIEW

File: log4j-core/src/main/java/org/apache/logging/log4j/core/net/SmtpManager.java [hunk 1]
```diff
@@ -166,9 +172,7 @@ private static String createManagerName(
         sb.append(isDebug ? ":debug:" : "::");
         sb.append(filterName);
 
-        final String hash = NameUtil.md5(sb.toString());
-        return "SMTP:" + hash;
-
+        return "SMTP:" + sb.toString();
```
Comment (vy) [inline log4j-core/src/main/java/org/apache/logging/log4j/core/net/SmtpManager.java:175]: Given `password` is in `sb`, this is not a good idea. Would you mind reverting this line, please?

Comment (sman-81) [reply log4j-core/src/main/java/org/apache/logging/log4j/core/net/SmtpManager.java:170]: Hello @vy,
thanks for your feedback.
I've removed password from creation of smtp manager name, as it is irrelevant for creating a unique per-configuration name/key (cannot have same user name with different password on same host).
Greetings,
Markus

**Comment by vy** [comment]
Thanks so much for the fix @sman-81!

**Comment by jvz** [comment]
@vy your cherry pick broke master. Can you take a look?

**Comment by vy** [comment]
[The associated GitHub Actions run](/apache/logging-log4j2/actions/runs/970729092) has succeeded. What is exactly broken?

**Comment by jvz** [comment]
Ralph figured it out earlier today. It was in the wrong package in master.

**Pull request merged**